osf/models/collection.py

```diff
@@ -31,6 +31,25 @@ class Meta:
     def _id(self):
         return self.guid._id
 
+    @classmethod
+    def _has_referent_perm(cls, guid, user, perm):
+        # TODO: Normalize permission checking to obviate this helper
```

Comment:
Do you mean that this will be obviated when django-guardian is used everywhere?

For the time being, you could define a common method across these types:


```python
class BaseFilenode:

    def has_permission(self, user, perm):
        return self.node and self.node.has_permisison(user, perm)


class Collection:

    def has_permission(self, user, perm):
        return user.has_perms(self.groups[perm], self)
```

Reply:
Either would work. If a permissions refactor is scheduled, though, I'd prefer to have one small and obviously-marked containment zone for bad practice, rather than have it scattered about. But it doesn't make much practical difference either way.

That said, I can change it if you think it's better that way.

Reply:
OK, I'm fine with this as is, as long as there's a plan to refactor it in the future. Is your team planning to do this?